### ISSUE TITLE
Prune mode in osx

### ISSUE THREAD

**Issue by ersinpw** [dashpay/dash#1787]
I am trying to switch the prune mode on.

I edited the file ~/Library/Application\ Support/DashCore/dash.conf to:
```
prune=945
txindex=0
```

This error pops up when I start DashCore:

> Enabling Masternode support requires turning on transaction indexing.Please add txindex=1 to your configuration and start with -reindex

However, when I set txindex to 0, this message appears:

> Prune mode is incompatible with -txindex.

I don't know how to -reindex using the GUI or terminal. How do I solve this issue?

**Comment by nmarley** [comment]
Hi, thanks for this! Setting the prune option and enabling the txindex aren't mutually compatible. If you need the full transaction index, you can't also enable pruning.

https://github.com/dashpay/dash/blob/20bacfab9b02e89df7b04c5270c7c7159349f6d3/src/init.cpp#L1005-L1014

But from the first message you posted, it looks like you also have `masternode` set, which requires the full transaction index:

https://github.com/dashpay/dash/blob/20bacfab9b02e89df7b04c5270c7c7159349f6d3/src/init.cpp#L1868-L1873


... so you can either run as a masternode (requires `txindex=1` also), or allow pruning, but not both.

I'm curious as to why you'd want to run a masternode on OSX (as this is 99% of the time a Macbook laptop and not OSX server, meaning it's not in a data center somewhere and has no uptime guarantees).

**Comment by nmarley** [comment]
So, just to clarify/update, apparently we don't fully support pruning, related: #1466, #1817

**Comment by codablock** [comment]
Maybe can be closed (as long as #1466 is kept open)?

**Comment by nmarley** [comment]
Closing as this is a duplicate of #1466, please track issue status there. This should be fixed once #2014 is merged in.
